lib/DxcSupport/WinAdapter.cpp

```diff
@@ -62,21 +62,40 @@ unsigned int SysStringLen(const BSTR bstrString) {
 }
 //===---------------------- Char converstion ------------------------------===//
 
-const char *CPToLocale(uint32_t CodePage) {
+const char *SetLocaleForCodePage(int Category, uint32_t CodePage) {
 #ifdef __APPLE__
```

Comment:
The stuff inside the `#ifdef __APPLE__` is starting to follow an extremely similar pattern to the others.  Could possibly be unified a little more here.

In fact, it's suspicious to me that the Apple one follows the same convention as Mariner for the UTF-8 locale, but not for the ISO88591-1 locale.  Wondering if that's working on Mariner by luck (eg it is failing but the default happens to work).

Reply:
| Platform | ACP | UTF8 |
|------|---|---|
| Apple | en_US.ISO8859-1 | en_US.UTF-8 |
| Mariner | en_US.iso88591 | en_US.UTF-8 |
| other | en_US.iso88591 | en_US.utf8 |

Reply:
I'm not 100% convinced this is working as expected.  On my Mariner:

```
$ locale -a
C
POSIX
en_US.UTF-8
```

And on Ubuntu:

```
$ locale -a
C
C.UTF-8
POSIX
en_US.utf8
```

Reply:
```
setlocale(LC_ALL, "en_US.iso88591");
setlocale(LC_ALL, "en_US.ISO8859-1");
```

Both return nullptr for me.

Reply:
For me.  Ubuntu:
```
en_US.UTF-8 ok!
en_US.utf8 ok!
en_US.ISO8859-1 bad.
en_US.iso88591 bad.
```

Mariner:

```
en_US.UTF-8 ok!
en_US.utf8 bad.
en_US.ISO8859-1 bad.
en_US.iso88591 bad.
```

So we could probably use en_US.UTF-8 everywhere.